### ISSUE TITLE
SPFX Teams app failed to get the token

### ISSUE THREAD

**Issue by Rtoribiog** [SharePoint/sp-dev-docs#7884]
### What type of issue is this?

other

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Target SharePoint environment

SharePoint Online

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [ ] 💥 Microsoft Edge
- [ ] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [X] other (enter in the "Additional environment details" area below)

### Additional environment details

- browser version   Teams desktop App
- SPFx version  1.13
- Node.js version   V 12.13
- etc

### Issue description

Related with issue   https://github.com/SharePoint/sp-dev-docs/issues/7824

We have spfx webpart embed in sharepoint page our intranet.  We created a teams app for the intranet page.

We updated already the manifest to use  https://{teamsitedomain}

the webpart is secured calling azure web app.

We enabled preview mode in teams and this is what we've seen in the authentication calls

https://xxx.sharepoint.com/_api/Microsoft.SharePoint.Internal.ClientSideComponent.Token.AcquireOBOToken?resource='89d57e6b-80df-4136-b9bd-18a817f2c1bc'&clientId='93fcfd5b-5de1-4bc4-90a4-860572bb4b89'

{"odata.error":{"code":"10009","message":{"lang":"en-US","value":"AADSTS500011: The resource principal named api://adf10e2b-b6e9-41d6-be2f-c12bb566019c/microsoft.spfx3rdparty.com was not found in the tenant named xxx. This can happen if the application has not been installed by the administrator of the tenant or consented to by any user in the tenant. You might have sent your authentication request to the wrong tenant.\r\nTrace ID: 1cac2d37-ddbb-49ad-9dd4-518d87921801\r\nCorrelation ID: 550933a0-a057-3000-caf0-9753cc712b06\r\nTimestamp: 2022-04-12 15:33:26Z"}}}

I've checked the manifest in sharepoint online client extensibility and this is what we have

![image](https://user-images.githubusercontent.com/687795/163011808-ac59998c-71df-4725-b479-b6bd5616d526.png)

https://xxx.sharepoint.com/_api/Microsoft.SharePoint.Internal.ClientSideComponent.Token.AcquireOBOToken?resource='https://xxx.sharepoint.com'&clientId='08e18876-6177-487e-b8b5-cf950c1e598c'

{"odata.error":{"code":"-1, System.AggregateException","message":{"lang":"en-US","value":"One or more errors occurred."}}}

right now it doesn't work in teams desktop and teams mobile ,  only works in teams web.

What can we do?

@GrahamMcMynn  @lucabandMSFT

**Comment by GrahamMcMynn** [comment]
hi @Rtoribiog  - sorry that you are experiencing this issue.

The reason this is happening is that our code did not update your identifier URI to be in the new format (api://<tenandId>/microsoft.spfx3rdparty.com).  This should happen when you visit the API access page with a global admin.  Have you done that?  Would it be possible to send me a trace of you loading the API access page (grahamc @ microsoft.com).

Basically that page should fix it up.  However, if that does not work, a way to ensure it fixes it properly is to delete the service principal in AAD and then revisit the API access page.

**Comment by Rtoribiog** [comment]
we visited the sharepoint admin api page.  Right now is working again in desktop app,  but it hasn't updated the sharepoint online client extensibility manifest. It has the same configuration

![image](https://user-images.githubusercontent.com/687795/164303128-b203b5d0-0929-40d2-ac5b-b6c2ff12de83.png)

**Comment by lucabandMSFT** [comment]
hey @Rtoribiog, thanks for the update.
Is the user that visiting the API Access page both **SP tenant admin and AAD Global Admin**? if yes, mind giving us some time availability? @GrahamMcMynn and I would like to have a screen call with you so we can see the issue happening and capture some real time traces.

Thanks

**Comment by Rtoribiog** [comment]
Only to clarify,  we haven't deleted the sharepoint app yet. it's still the old one but we visited the sharepoint admin page as sharepoint admin not as aad global admin.  Let me try again and i will let you know.  Thanks

**Comment by Rtoribiog** [comment]
ok,  we tested right now and it updated the identifierUris,  also updated accessTokenAcceptedVersion: 1.
For desktop version is working again, but for mobile is not working.  It shows an spinner first blue and later violet,  but nothing show up..
